### DISCUSSION TITLE
Varying number of "OR" conditions possible?

### DISCUSSION THREAD

**Discussion by Freem11** [supabase/supabase#13305] [Questions]
I need o create a function that will have a varying number of OR conditions 

user choses animals and i want to see if they are in the db, they can chose 1 to x number of animals from the list

is there any way to get supabase's postgres SQL to accept multiple, but changing in number of OR conditions 

example: 
chose 2 animals
select * from table where animal=animal1 or animal=animal2

chose 4 animals 
select * from table where animal=animal1 or animal=animal2 or animal=animal3 or animal=animal4

I used to do this via javascript to build our the SQL statement by looping then run it from my own express server, but i cant seem to hack a solution to this problem using supabase functions 

i have been working on this for days now so anyone who can solve this will be a hero to me

**Comment by GaryAustin1** [comment]
Why not just the `in` filter. https://supabase.com/docs/reference/javascript/in
It matches a single value column to any value in an array of values.

**Reply by Freem11** [reply]
Hi Gary
I had to do a group by so I needed to do a custom function, but I found the SQL equivalent: "ANY" which did the trick, only thing I had to do after that was then do a subquery so that I could pull info by criteria but the only group by a the ones I really needed..sigh 2 days of bashing my head against it but its finally working
Thanks for your suggestion though for more standard queries ill give that a try next time
